src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -602,6 +602,10 @@ CTranslatorDXLToPlStmt::TranslateDXLTblScan(
 		plan->qual = qual;
 	}
 
+	if (md_rel->IsAORowOrColTable())
+	{
+		CheckSafeTargetListForAOTables(plan->targetlist);
```

Comment:
I guest we need use the `RelationIsNonblockRelation` in this case?

The `PAX` am also not allow `System column` in target list.